presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -0,0 +1,251 @@
+
+        for (TableStatisticType type : statisticsMetadata.getTableStatistics()) {
+            if (type != ROW_COUNT) {
+                throw new PrestoException(NOT_SUPPORTED, "Table-wide statistic type not supported: " + type);
+            }
+        }
+
+        FunctionRegistry functionRegistry = metadata.getFunctionRegistry();
+        if (!statisticsMetadata.getTableStatistics().isEmpty()) {
+            QualifiedName count = QualifiedName.of("count");
+            AggregationNode.Aggregation aggregation = new AggregationNode.Aggregation(
+                    new FunctionCall(count, ImmutableList.of()),
+                    functionRegistry.resolveFunction(count, ImmutableList.of()),
+                    Optional.empty());
+            Symbol symbol = symbolAllocator.newSymbol("rowCount", BIGINT);
+            aggregations.put(symbol, aggregation);
+            descriptor.addTableStatistic(symbol, ROW_COUNT);
+        }
+
+        ImmutableMap.Builder<Symbol, Expression> projections = ImmutableMap.builder();
+        for (ColumnStatisticMetadata columnStatisticMetadata : statisticsMetadata.getColumnStatistics()) {
+            String columnName = columnStatisticMetadata.getColumnName();
+            ColumnStatisticType statisticType = columnStatisticMetadata.getStatisticType();
+            Symbol inputSymbol = columnToSymbolMap.get(columnName);
+            verify(inputSymbol != null, "inputSymbol is null");
+            Type inputType = requireNonNull(symbolAllocator.getTypes().get(inputSymbol), "inputType is null");
+            SingleColumnStatisticAggregation aggregation = createColumnAggregation(statisticType, inputSymbol, inputType);
+            Symbol symbol = symbolAllocator.newSymbol(statisticType + ":" + columnName, aggregation.getOutputType());
+            aggregations.put(symbol, aggregation.getAggregation());
+            descriptor.addColumnStatistic(symbol, columnStatisticMetadata);
+            aggregation.getInputProjection().ifPresent(projections::put);
+        }
+
+        StatisticAggregations aggregation = new StatisticAggregations(aggregations.build(), groupingSymbols);
+        return new CreateStatisticAggregationsResult(aggregation, descriptor.build(), projections.build());
+    }
+
+    private SingleColumnStatisticAggregation createColumnAggregation(ColumnStatisticType statisticType, Symbol input, Type inputType)
+    {
+        switch (statisticType) {
+            case MIN: {
+                checkArgument(inputType.isOrderable(), "Input type is not orderable: %s", inputType);
+                return createAggregation(QualifiedName.of("min"), input.toSymbolReference(), inputType, inputType);
+            }
+            case MAX: {
+                checkArgument(inputType.isOrderable(), "Input type is not orderable: %s", inputType);
+                return createAggregation(QualifiedName.of("max"), input.toSymbolReference(), inputType, inputType);
+            }
```

Comment:
nit: these `{`, `}` are redundant in most of the cases and affect readability. IMHO it's better without them, but no strong opinion here

Reply:
I usually go without `{ }` if the switch case blocks are one-liners. I like them for multiliners though.